zkevm-circuits/src/evm_circuit.rs

```diff
@@ -1 +1,918 @@
+
+        // fixed_table contains pre-built tables identified by tag including:
+        // - different size range tables
+        // - bitwise table
+        // - comparator table
+        // - ...
+        let fixed_table = [
+            meta.fixed_column(), // tag
+            meta.fixed_column(),
+            meta.fixed_column(),
+            meta.fixed_column(),
+        ];
+
+        let (
+            qs_op_execution,
+            qs_byte_lookups,
+            op_execution_state_curr,
+            op_execution_state_next,
+            op_execution_free_cells,
+        ) = Self::configure_allocations(meta, q_step, qs_byte_lookup, advices);
+
+        // independent_lookups collect lookups by independent selectors, which
+        // means we can sum some of them together to save lookups.
+        let mut independent_lookups =
+            Vec::<(Expression<F>, Vec<Lookup<F>>)>::new();
+
+        let op_execution_gadget = OpExecutionGadget::configure(
+            meta,
+            r,
+            qs_op_execution,
+            qs_byte_lookups,
+            op_execution_state_curr.clone(),
+            op_execution_state_next,
+            op_execution_free_cells,
+            &mut independent_lookups,
+        );
+
+        // TODO: call_initialization_gadget
+
+        Self::configure_lookups(
+            meta,
+            qs_byte_lookup,
+            advices,
+            fixed_table,
+            rw_table,
+            op_execution_state_curr,
+            independent_lookups,
+        );
+
+        EvmCircuit {
+            q_step,
+            qs_byte_lookup,
+            fixed_table,
+            rw_table,
+            op_execution_gadget,
+        }
+    }
+
+    // TODO: refactor return type
+    #[allow(clippy::type_complexity)]
+    fn configure_allocations(
+        meta: &mut ConstraintSystem<F>,
+        q_step: Selector,
+        qs_byte_lookup: Column<Advice>,
+        advices: [Column<Advice>; CIRCUIT_WIDTH],
+    ) -> (
+        Expression<F>,
+        Vec<Cell<F>>,
+        OpExecutionState<F>,
+        OpExecutionState<F>,
+        Vec<Cell<F>>,
+    ) {
+        let mut qs_byte_lookups = Vec::with_capacity(CIRCUIT_HEIGHT);
+        meta.create_gate("Query byte lookup as a synthetic selector", |meta| {
+            for h in 0..CIRCUIT_HEIGHT as i32 {
+                qs_byte_lookups.push(Cell {
+                    expression: meta.query_advice(qs_byte_lookup, Rotation(h)),
+                    column: qs_byte_lookup,
+                    rotation: h as usize,
+                });
+            }
+
+            vec![0.expr()]
+        });
+
+        // TODO: Enable this when switch to kzg branch and remove the failure
+        //       VerifyFailure::ConstraintPoison
+        // meta.create_gate("Query byte lookup as a synthetic selector", |meta| {
+        //     let expr = meta.query_advice(qs_byte_lookup, Rotation::cur());
+
+        //     vec![expr.clone() * (Expression::Constant(F::one()) - expr)]
+        // });
```

Comment:
Nitpick: let's move this to a new issue. Avoid commit commented code.

```suggestion
```

Reply:
Fixed in [`8063d88`](https://github.com/appliedzkp/zkevm-circuits/commit/8063d888c9db7e47b9f3e8088570048f0f8201a6).